float64 frame_x
float64 frame_y
bool dFrame_calculated
